featuregates/client/pkg/util/featuregate.go

```diff
@@ -82,7 +86,68 @@ func ComputeFeatureStates(featureGateSpec configv1alpha1.FeatureGateSpec, featur
 	return activated, deactivated, unavailable
 }
 
+// ComputeFeatureStatesForCoreAPIGroupResources takes a corev1alpha2 FeatureGate spec and computes the actual state (activated, deactivated or unavailable)
+// of the features in the gate by referring to a list of Feature resources.
+func ComputeFeatureStatesForCoreAPIGroupResources(featureGateSpec corev1alpha2.FeatureGateSpec, features []corev1alpha2.Feature) (activated, deactivated, unavailable []string) {
+	// Collect features to be activated/deactivated in the spec.
+	toActivate := sets.String{}
+	toDeactivate := sets.String{}
+	for _, f := range featureGateSpec.Features {
+		if f.Activate {
+			toActivate.Insert(f.Name)
+		} else {
+			toDeactivate.Insert(f.Name)
+		}
+	}
+
+	// discovered is set a set of available features that are discoverable.
+	discovered := sets.String{}
+	// discoveredDefaultActivated is a set of available features that are discoverable and activated by default.
```

Comment:
Nit: 105 and 106 also seem too long as well as comments in the rest of this doc.